### PR TITLE
Add replacements for missing FPYs in depletion chains, several changes for CASL chain

### PR DESCRIPTION
**Pull request by paulromano**
This PR modifies our depletion chains to include FPYs for nuclides that do not have any (e.g., Pu243) using the new capability from openmc-dev/openmc#1586. There are also a number of fixes in the CASL depletion chain:
- For fast reactor calculations, often the fuel has Zr and/or Mo in it. I've added several isotopes of Zr and Mo in the CASL chain to capture transmutation in these elements appropriately.
- I've changed the strategy in the CASL chain generation when reaction products are missing. The biggest effect is for (n,p) reactions, which often result in products that beta- decay right back to where they started.

Sorry @drewejohnson to request yet another review from you. I owe you a :beer: for sure!

### CODE REVIEW

File: depletion/generate_endf71_chain_casl.py [hunk 1]
```diff
@@ -144,8 +174,9 @@ def main():
                         chain.reactions.append(name)
 
                     if daughter not in decay_data:
-                        missing_rx_product.append((parent, name, daughter))
-                        daughter = 'Nothing'
+                        daughter = replace_missing(daughter, decay_data, all_decay_data)
```
Comment (drewejohnson) [inline depletion/generate_endf71_chain_casl.py:177]:
```suggestion
                        daughter = replace_missing_decay_product(
                            daughter, decay_data, all_decay_data)
```

File: depletion/generate_endf71_chain_casl.py [hunk 2]
```diff
@@ -30,6 +30,33 @@
...
+            break
+
+        decay_obj = all_decay_data[product]
+        if decay_obj.nuclide['stable'] or decay_obj.half_life.n > 24*60*60:
+            product = None
+            break
+
+        dominant_mode = max(decay_obj.modes, key=lambda x: x.branching_ratio)
+        if len(decay_obj.modes) > 1:
+            print(product, decay_obj.modes)
```
[the remark this answers is not in the document]
Comment (paulromano) [reply depletion/generate_endf71_chain_casl.py:53]: Oops!

File: depletion/generate_endf71_chain_casl.py [hunk 3]
```diff
@@ -30,6 +30,33 @@
...
+    while product not in decay_data:
+        # Missing products
+        if product not in all_decay_data:
+            product = None
+            break
+
+        decay_obj = all_decay_data[product]
+        if decay_obj.nuclide['stable'] or decay_obj.half_life.n > 24*60*60:
+            product = None
+            break
```
[the remark this answers is not in the document]
Comment (paulromano) [reply depletion/generate_endf71_chain_casl.py:49]: Almost -- this condition is for when a product being searched for is _not_ listed in the chain but is stable or doesn't have a short half-life. If it's stable and not in the chain, well then you obviously can't do anything else. If the half-life is not short, then you do have to make some sort of ad hoc cutoff; if you allow it to continue decaying to whatever the next product is, you'll start to build up some nuclide much faster than it should be building up.

And apologies -- I should have put more comments in the code (will do that now)

[the remark this answers is not in the document]
Comment (drewejohnson) [reply depletion/generate_endf71_chain_casl.py:53]: Thanks for the clarification!

**Pull request merged**